Enhance Clarity of Version Scheduling Interface

KnollElias opened the pull request:
This pull request endeavors to clarify the Version Scheduling feature by presenting clearer descriptions and info boxes to users. The modifications are aimed at ensuring a more intuitive user experience when scheduling versions for publication.

Technical Details: 
- Reorganized the logic to set descriptive texts and info messages based on whether a scheduled version exists or an active version exists.
- Introduced additional HTML structure to accommodate alternative texts and info messages.
- Added CSS styling for improved visibility and user interaction with the info boxes.

Screenshots:
![Bildschirmfoto 2023-10-27 um 08 22 00](https://github.com/concretecms/concretecms/assets/107244530/e5413930-c53c-4036-a8e4-f1f757e0184e)

Additional Context:
This is my second pull request, inspired by feedback from clients using our Concrete CMS sites. The changes have been tested for consistency with the project's coding style guidelines by running the style fix.

KnollElias committed:
Merge pull request #2 from concretecms/9.3.x

9.3.x

KnollElias committed:
Merge pull request #3 from concretecms/9.3.x

9.3.x

KnollElias committed:
Update schedule.php

KnollElias committed:
Merge branch 'develop' of https://github.com/KnollElias/concretecms into develop

KnollElias committed:
Update schedule.php

KnollElias committed:
Update schedule.php

mlocati commented on concrete/elements/pages/schedule.php:
```diff
@@ -38,50 +38,93 @@
 $timezone = $dateService->getTimezoneDisplayName($timezone);
+
+// activeVersionExists
+$activeVersionExistsText = t('At the moment, the existing live version will remain.');
```
Why do you define these variables? I'd instead simply write the strings where they are needed...
For example, instead of having:

```php
$activeVersionExistsText = t('At the moment, the existing live version will remain.');
$primaryText = $activeVersionExistsText;
```

I'd simply prefer

```php
$primaryText = t('At the moment, the existing live version will remain.');
```

Also, some of the translatable strings are missing `t()`.

KnollElias replied:
Hello Sir,

From a maintainability standpoint, I initially thought that defining everything in one place would be easier. However, given that the if-statement on line 69 `if($scheduledVersionExists == false) {` is self-explanatory, I agree that it is probably a better idea to define the strings directly as you suggested. 

Is there anything else that could be optimized? Also, regarding the color code `#0099ff` in the `.svg-icon` class, is it defined somewhere within the project where I could reference it easily?

KnollElias committed:
upgraded schaedule variable declaration

aembler reviewed (changes requested):
Looks interesting! We should be able to do all this without any new inline styles. Just use Font awesome for your icons. I don't want to have custom CSS here. Also I agree with mlocati there's no reason to have the strings live in a separate spot, just put them inline wherever they're being printed out.

KnollElias committed:
Update schedule.php

KnollElias commented:
I’ve replaced all custom SVGs with Font Awesome icons and removed inline styles in favor of minimal, class-based CSS essential for the toggle functionality. Additionally, text strings are now placed directly where needed.

KnollElias commented:
I wanted to check if there's anything else needed on my end for PR #11731. I have implemented the changes you suggested and am wondering if there are any additional adjustments required. Your guidance on any further steps would be greatly appreciated.

KnollElias commented:
@aembler @mlocati Could you please take another look at the pull request #11731 and let me know if there is anything else I should adjust or improve? Your feedback is very valuable to me. Thank you for your time.

mlocati commented on concrete/elements/pages/schedule.php:
```diff
@@ -41,51 +41,68 @@
 </div>
 
 <style type="text/css">
+    .text-active, .info-active {
```
Will this rule be applied to any element in the page?

KnollElias committed:
Refined CSS scope for scheduling interface

KnollElias commented:
Thanks for the feedback. Scoped CSS with #version-scheduling to target relevant elements and avoid conflicts. Ready for another review – open to more suggestions or tasks!

KnollElias committed:
Update schedule.php

KnollElias committed:
Remove Custom CSS schedule

KnollElias commented:
@mlocati Removed custom CSS with Bootstrap but now JS is required.

mlocati commented on concrete/elements/pages/schedule.php:
```diff
@@ -41,52 +42,51 @@
+$(document).ready(function(){
+    $('#keepOtherScheduling').change(function(){
+        if(this.checked) {
+            $('.text-standard, .info-standard').addClass('d-none');
```
This alters every element with `text-standard` or `info-standard` classes, even those we may not want to touch (that is, something else on the page).

What about

```js
$('#version-scheduling .text-standard, #version-scheduling .info-standard').addClass('d-none');
```

that way, we'll only touch the elements that are children of the element with id `version-scheduling`.

mlocati replied:
(of course, the same applyes to the lines below)

KnollElias replied:
Thanks for the feedback, @mlocati. I've scoped the CSS to target elements under #version-scheduling, preventing conflicts. Please let me know if there are more changes needed. Thanks!

marcokuoni replied:
@KnollElias Can you also scope the add and removeClass functions?

KnollElias committed:
Update schedule specific classes

marcokuoni commented:
@aembler What do you think to this PR, are we missing something?

aembler commented:
I find these messages less intuitive than they used to be, but I don't have a great answer. I think it's best to approve these and try to improve them in the future.
